### PR TITLE
Fix logger formatting

### PR DESCRIPTION
**Pull request by avsej**
Motivation
----------
Current implementation does not allow to specify any formatting. It
even cannot output log level, because `writeln()` accepts
`varags[string]` and silently ignores everything before last argument.

Modification
------------
Perform formatting in the single place during substitution. Make log
level optional as all other substitution variables. Also make verbose
logging more informative and parseable.

Result
------
Correct handling formatting and substitutions. Machine-friendly default
output of verbose logger.

### CODE REVIEW

File: lib/pure/logging.nim [hunk 1]
```diff
@@ -87,12 +92,9 @@ type
...
     PFileLogger: FileLogger, PRollingFileLogger: RollingFileLogger].}
 
-proc substituteLog(frmt: string): string =
-  ## converts $date to the current date
-  ## converts $time to the current time
-  ## converts $app to getAppFilename()
-  ## converts
-  result = newStringOfCap(frmt.len + 20)
+proc substituteLog(frmt: string, level: Level, args: varargs[string, `$`]): string =
+  var msg = join(args)
```
Comment (Araq) [inline lib/pure/logging.nim]: Better avoid that 'join' call for performance. Instead iterate over ther args array twice: One time for the length calculations, and then a second time to append to result.

File: lib/pure/logging.nim [hunk 2]
```diff
@@ -19,7 +19,12 @@
 ## ============  =======================
 ## $date         Current date
 ## $time         Current time
+## $datetime     $dateT$time
```
Comment (Araq) [inline lib/pure/logging.nim]: Typo? 'T'?

Comment (avsej) [inline lib/pure/logging.nim]: no, T is separator between date and time

Comment (avsej) [inline lib/pure/logging.nim]: This is the reason of `$datetime`, because `$dateT$time` just does not work

File: lib/pure/logging.nim [hunk 3]
```diff
@@ -87,12 +92,13 @@ type
...
     PFileLogger: FileLogger, PRollingFileLogger: RollingFileLogger].}
 
-proc substituteLog(frmt: string): string =
-  ## converts $date to the current date
-  ## converts $time to the current time
-  ## converts $app to getAppFilename()
-  ## converts
-  result = newStringOfCap(frmt.len + 20)
+proc substituteLog(frmt: string, level: Level, args: varargs[string, `$`]): string =
+  var msg = ""
```
Comment (Araq) [inline lib/pure/logging.nim]: The point is to get rid of this temporary 'msg' string and the allocations it causes. ;-)

Comment (avsej) [inline lib/pure/logging.nim]: ok

**Pull request merged**